UnboundLib/GameModes/GameModeManager.cs

```diff
@@ -15,6 +16,10 @@ public static class GameModeManager
         private static Dictionary<string, List<Func<IGameModeHandler, IEnumerator>>> hooks = new Dictionary<string, List<Func<IGameModeHandler, IEnumerator>>>();
         private static Dictionary<string, List<Func<IGameModeHandler, IEnumerator>>> onceHooks = new Dictionary<string, List<Func<IGameModeHandler, IEnumerator>>>();
 
+        // public properties that return deep copies of the handlers and gameModes dictionarys (the values in the dictionaries returned are shallow copies)
+        public static Dictionary<string, IGameModeHandler> Handlers => handlers.ToDictionary(kv => kv.Key, kv => kv.Value);
```

Comment:
It's possible to protect the dictionaries more by returning them as a `ReadOnlyDictionary` (from `System.Collections.ObjectModel`). Something akin to

```CSHARP
public static ReadOnlyDictionary<string, IGameModeHandler> Handlers
{
    get
    {
        return new ReadOnlyDictionary<string, IGameModeHandler>(handlers.ToDictionary(kv => kv.Key, kv => kv.Value));
    }
}
```

Reply:
Good idea, added in 3e60ff0ca1c54e43cdba691af517b17257c34594